### ISSUE TITLE
Partially blurring nodes

### ISSUE THREAD

**Issue by glibvana** [konvajs/konva#998]
I have 2 layers. Background layer is an image, and an additional layer on top for various konvajs shapes.

I would like to draw a rectangle on the top layer that will blur everything in the background that it is covering.

Example code:

`import {Stage, Layer, FastLayer, Rect, Filters} from "konva";

const stage = new Stage({
    container: 'canvas',
    width: this.canvasWidth,
    height: this.canvasHeight
});

const shapeLayer = new Layer({
    name: 'shapes'
})

const imageLayer = new FastLayer({
    name: 'image'
});

stage.add(imageLayer);
stage.add(shapeLayer);

imageLayer.moveToBottom();

const rect = new Rect({
    name: 'Rect',
    x: 1,
    y: 1,
    width: 200,
    height: 200,
    draggable: true,
    opacity: 0.5,
    fill: '#000'
});

rect.cache();
rect.filters([Filters.Blur]);
rect.blurRadius(20);
shapeLayer.draw();`

This does not work however. I only see a black transparent rectangle without any of the background image blurred in anyway. Tried the same thing with the 'Pixelate' filter. What is being done incorrectly, and what is the correct way to do this?

Thanks in advance

**Comment by lavrton** [comment]
> I would like to draw a rectangle on the top layer that will blur everything in the background that it is covering

I don't think it is possible. Filters only affect nodes where they are applied.  So blurred rectangle will not blur everything under it.

You can do these:
- Cache the whole layer and apply filters to it.
- Or try to use CSS filters manually on the stage container. CSS filters will be not included in exports (such as `stage.toDataURL()`).
